src/functions/authoriser/framework/handler.ts

```diff
@@ -13,21 +56,25 @@ const newCustomAuthorizerResult = (principalId: string, effect: string, resource
     },
   });
 
-export async function handler(event: CustomAuthorizerEvent, fnCtx: Context)
-  : Promise<CustomAuthorizerResult> {
-  const token = event.authorizationToken;
-  const userId = 'user-id';
-  if (token) {
-    switch (token.toLowerCase()) {
-      case 'allow':
-        return newCustomAuthorizerResult(userId, 'Allow', event.methodArn);
-      case 'deny':
-        return newCustomAuthorizerResult(userId, 'Deny', event.methodArn);
-      case 'unauthorized':
-        throw Error('unauthorized');
-      default:
-        throw Error('unauthorized');
-    }
+/**
+ * Exported entry point to the Custom Authorizer Lambda.
+ */
+let adJwtVerifier: AdJwtVerifier | null = null;
+export async function handler(event: CustomAuthorizerEvent) : Promise<CustomAuthorizerResult> {
+  // One-time initialization
+  if (adJwtVerifier === null) {
+    adJwtVerifier = await createAdJwtVerifier();
+  }
+
+  // Get token from event, and verify
+  const token = event.authorizationToken || '';
+  ensureNotNullOrEmpty(token, 'event.authorizationToken');
```

Comment:
ensureNotNullOrEmpty() only throws an error which would probably make Lambda respond with 500 if the token is null or empty
I'd rather expect that Lambda to only respond with 200 or 403 (500 if the code itself is somehow broken)
...unless I'm wrong because I'm not a developer ;)

Reply:
If the token is empty, API gateway doesn't actually even invoke the auth lambda.  It responds with a `401 Unauthorized` immediately, because it knows the token is invalid.  The check here is just useful in debugging and testing.